### ISSUE TITLE
user activation from email bugs

### ISSUE THREAD

**Issue by rjvillanueva89** [benedmunds/CodeIgniter-Ion-Auth#1438]
**Which branch are you using?**
4 (branch used should match CodeIgniter version)

**What commit hash are you on?**
fe132df76952efda6ae74b71fdf33cae76e2725e

**What CodeIgniter version are you using?**
v4

**What PHP version are you using?**
PHP 7.0

**Post your Ion Auth config below**
```
public $emailActivation = true;
public $manualActivation = true;
```

**Describe the bug**
errors when activating user from email click here to activate; also admin not logged in;

on **ion-auth/Controller/Auth.php line 451:**
`if (! $code)` 

if code is present would skip the activation code check and default to admin check

on **ion-auth/Models/IonAuthModel.php line 385:**
`public function activate($id, bool $code=false)`

because of the bool var declaration; when passing the code it converts the value to bool value and the code isn't being broken down properly on retrieveSelectorValidatorCouple function to get the selector and validator

**To Reproduce**
1. create account 
2. activate account from email (/auth/activate/id/code)

**Expected behavior**
on **ion-auth/Controllers/Auth.php line 455**
`$this->ionAuth->isAdmin()`
would cause an error

**Comment by benedmunds** [comment]
This definitely makes sense, thanks for reporting it.  Could you send a PR with the fix?

**Comment by rjvillanueva89** [comment]
Sure thing.

there is also an issue on **ion-auth/Models/IonAuthModel.php line 393**
`if ($code === false || ($user && $user->id === $id))`
that is causing an error so the activate request is returning as activate_unsuccessful

**Comment by rjvillanueva89** [comment]
I've just checked out the repo and just saw the updates and were already fixed on commit 09cfbd07220ec96ca932941404b0fd17b8ba976f. Sorry about that.

**Comment by benedmunds** [comment]
Ah awesome, thanks!
